lib/metrics.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const {loggers: log, addon: aob} = require('.');
+const v8 = require('v8');
+
+class Metrics {
+  constructor (options = {}) {
+    this.id = undefined;
+    this.interval = options.interval || 60 * 1000;
+    this.prefix = options.prefix || 'trace.node';
+    this.metrics = [];
+    this.state = 'initial';
+  }
+
+  start () {
+    if (this.state !== 'started') {
+      // start the metrics gathered by the C++ code
+      aob.metrics.start();
+
+      this.id = setInterval(() => {
+        this.reportMetrics();
+      }, this.interval);
+      // don't let metrics keep the process alive
+      this.id.unref();
+    }
+    return this.state = 'started';
+  }
+
+  stop () {
+    if (this.state !== 'started') {
+      return;
+    }
+    aob.metrics.stop();
+    clearInterval(this.id);
+    return this.state = 'stopped';
+  }
+
+  getState () {
+    return this.state;
+  }
+
+  resetInterval (interval) {
+    this.interval = interval;
+    this.stop();
+    this.start();
+  }
+
+  addMetricV (metric, value, n = 1) {
+    this.metrics.push({
+      name: `${this.prefix}.${metric}`,
+      count: n,
+      addHostTag: true,
+      value,
+    });
+  }
+
+  addMetricI (metric, n = 1) {
```

Comment:
can't find addMetricI being used anywhere

Reply:
nope, it's not. but it's there for completeness so i don't have to go look it up if i end up needing it.